Replace aframe-teleport-controls with bespoke teleporter.

johnshaughnessy opened the pull request:
This brings `aframe-teleport-contols` into our project, removes all the pieces we don't use, and changes it to poll `userinput`. I initially was just trying to track down a tricky issue I was facing when trying to remove `tracked-controls`, but I want to do this anyway to remove some code and uses of `action-to-event`.

netpro2k commented on src/components/teleporter.js:
```diff
@@ -0,0 +1,377 @@
+      this.timeTeleporting = 0;
+      this.hit = false;
+      this.rayCurve.mesh.visible = true;
+      this.rayCurve.mesh.updateMatrixWorld();
```
Realize this is probably copypaste from teleport controls, but unclear why this needs to get updated

netpro2k commented on src/components/teleporter.js:
```diff
@@ -0,0 +1,377 @@
+    }
+
+    if (userinput.get(confirm)) {
+      this.hitEntity.setAttribute("visible", false);
```
```suggestion
      this.hitEntity.object3D.visible = false;
```

netpro2k commented on src/components/teleporter.js:
```diff
@@ -0,0 +1,377 @@
+    this.rayCurve.material.color.set(color);
+    this.rayCurve.material.opacity = opacity;
+
+    this.hitEntity.setAttribute("visible", this.hit);
```
```suggestion
    this.hitEntity.object3D.visible = this.hit;
```

netpro2k commented on src/components/teleporter.js:
```diff
@@ -0,0 +1,377 @@
+
+    this.hitEntity.setAttribute("visible", this.hit);
+    if (this.hit) {
+      this.hitEntity.setAttribute("position", this.hitPoint);
```
```suggestion
      this.hitEntity.object3D.position = this.hitPoint;
```

netpro2k commented on src/components/teleporter.js:
```diff
@@ -0,0 +1,377 @@
+    if (this.hit) {
+      this.hitEntity.setAttribute("position", this.hitPoint);
+      this.hitEntity.object3D.traverse(o => {
+        o.matrixNeedsUpdate = true;
```
This seems really surprising. The matrix updating code should handle updating children itself, you should only need to set it on hitEntity

netpro2k commented on src/components/teleporter.js:
```diff
@@ -0,0 +1,377 @@
+      const dRadii = this.data.outerRadius - this.data.hitCylinderRadius;
+      const outerScale = (this.data.outerRadius - easeIn(percentToDraw) * dRadii) / this.data.outerRadius;
+      this.outerTorus.object3D.scale.set(outerScale, outerScale, 1);
+      this.torus.setAttribute("material", "opacity", hitEntityOpacity);
```
would like to get rid of use of the material component here, but might make sense as its own PR

netpro2k commented on src/components/teleporter.js:
```diff
@@ -0,0 +1,377 @@
+    this.el.emit(sound);
+  },
+
+  createHitEntity() {
```
As mentioned above, rewriting this to just use Object3Ds would be nice, especially since using the position/rotation/scale components is extra scary with our matrixUpdate changes... That said teleport-controls was mostly working so probably fine for that to be a second pass if it turns out to be not a quick change.

netpro2k commented on src/components/teleporter.js:
```diff
@@ -0,0 +1,377 @@
+    start: { type: "string" },
+    confirm: { type: "string" },
+    speed: { default: 12 },
+    collisionEntities: { default: "" },
```
We only care about navmesh and we end up having a whole component dedicated to getting this to look at navmesh at the right time, I would vote we just do it more directly and less configurable.

netpro2k commented on src/components/teleporter.js:
```diff
@@ -0,0 +1,377 @@
+const CYLINDER_TEXTURE =
+  "data:image/png;base64,iVBORw0KGgoAAAANSUhEUgAAAAEAAAAQCAYAAADXnxW3AAAACXBIWXMAAAsTAAALEwEAmpwYAAAKT2lDQ1BQaG90b3Nob3AgSUNDIHByb2ZpbGUAAHjanVNnVFPpFj333vRCS4iAlEtvUhUIIFJCi4AUkSYqIQkQSoghodkVUcERRUUEG8igiAOOjoCMFVEsDIoK2AfkIaKOg6OIisr74Xuja9a89+bN/rXXPues852zzwfACAyWSDNRNYAMqUIeEeCDx8TG4eQuQIEKJHAAEAizZCFz/SMBAPh+PDwrIsAHvgABeNMLCADATZvAMByH/w/qQplcAYCEAcB0kThLCIAUAEB6jkKmAEBGAYCdmCZTAKAEAGDLY2 ...
```
Might make this an external image
![image](https://user-images.githubusercontent.com/130735/53534684-c713d200-3ab4-11e9-8912-87619d09b46a.png)

johnshaughnessy commented:
Thanks for the feedback @netpro2k -- I'll aim to have a fast followup addressing these comments. For now I'm going to push as-is because I want this and https://github.com/mozilla/hubs/pull/970 in more than I want to fix these.
